// @ezcater/recipe/lib/components/EzCalendar/EzCalendar.styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CalendarTable = exports.Day = exports.Row = exports.WeekdayName = exports.MonthName = exports.MonthNavigation = void 0;

var _core = require("@emotion/core");

var _styled = _interopRequireDefault(require("../../themes/styled"));

var MonthNavigation =
/*#__PURE__*/
(0, _styled["default"])("span", {
  target: "e243hl70"
})("flex:0;white-space:nowrap;");
exports.MonthNavigation = MonthNavigation;
var MonthName =
/*#__PURE__*/
(0, _styled["default"])("span", {
  target: "e243hl71"
})("font-weight:", function (props) {
  return props.theme.fontWeights.bold;
}, ";");
exports.MonthName = MonthName;
var WeekdayName =
/*#__PURE__*/
(0, _styled["default"])("span", {
  target: "e243hl72"
})("padding:", function (props) {
  return props.theme.spacing.sm;
}, " 0;font-weight:", function (props) {
  return props.theme.fontWeights.bold;
}, ";");
exports.WeekdayName = WeekdayName;
var Row =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "e243hl73"
})("display:flex;align-items:center;> *{flex:1;text-align:center;}");
exports.Row = Row;

var day = function day(_ref) {
  var theme = _ref.theme;
  return (
    /*#__PURE__*/
    (0, _core.css)("button{border:", theme.borderWidth[0], " solid transparent;border-radius:", theme.borderRadius[1], ";&[aria-disabled='true']{opacity:0.6;cursor:default;color:", theme.colors.interactive.disabled.foreground, ";pointer-events:none;}}")
  );
};

var selected = function selected(_ref2) {
  var theme = _ref2.theme,
      isSelected = _ref2.isSelected;
  return isSelected &&
  /*#__PURE__*/
  (0, _core.css)("button{border-radius:", theme.borderRadius[1], ";border:", theme.borderWidth[0], " solid ", theme.colors.interactive.checked.border, ";background-color:", theme.colors.interactive.checked.background, ";}");
};

var Day =
/*#__PURE__*/
(0, _styled["default"])("span", {
  target: "e243hl74"
})(day, selected);
exports.Day = Day;
var CalendarTable =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "e243hl75"
})("width:100%;button{padding:", function (props) {
  return props.theme.spacing.xs;
}, ";width:100%;font-family:inherit;font-size:inherit;font-weight:", function (props) {
  return props.theme.fontWeights.bold;
}, ";color:", function (props) {
  return props.theme.colors.interactive.base;
}, ";appearance:none;border:0;background-color:transparent;cursor:pointer;}");
exports.CalendarTable = CalendarTable;
