// @ezcater/recipe/lib/components/EzNavigation/EzNavigation.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _en = _interopRequireDefault(require("./en"));

var _Logo = _interopRequireDefault(require("./Logo"));

var _Menu = _interopRequireDefault(require("./Menu"));

var _Hamburger = _interopRequireDefault(require("./Hamburger"));

var _UserMenu = _interopRequireDefault(require("./UserMenu"));

var _Notifications = require("./Notifications");

var _EzNavigation = require("./EzNavigation.styles");

var _hooks = require("../../utils/hooks");

var countNotifications = function countNotifications(links) {
  return links.reduce(function (sum, link) {
    if (!link.notifications) return sum;
    var count = Number.isInteger(link.notifications) ? Number(link.notifications) : 0;
    return sum + count;
  }, 0);
};

var EzNavigation = function EzNavigation(_ref) {
  var children = _ref.children,
      links = _ref.links,
      utilityLinks = _ref.utilityLinks,
      userMenu = _ref.userMenu,
      _ref$home = _ref.home,
      logo = _ref$home.logo,
      homeLink = (0, _objectWithoutProperties2["default"])(_ref$home, ["logo"]);

  var _useTranslation = (0, _hooks.useTranslation)(_en["default"]),
      t = _useTranslation.t;

  var _useState = (0, _react.useState)(true),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      hidden = _useState2[0],
      setHidden = _useState2[1];

  var handleSidebarToggle = function handleSidebarToggle() {
    return setHidden(function (s) {
      return !s;
    });
  };

  var notificationSummary = countNotifications(links) + countNotifications(utilityLinks || []);
  return _react["default"].createElement(_EzNavigation.Wrapper, null, _react["default"].createElement(_EzNavigation.NavWrapper, {
    opened: hidden
  }, _react["default"].createElement(_Hamburger["default"], {
    opened: !hidden,
    onClick: handleSidebarToggle
  }), notificationSummary > 0 && _react["default"].createElement(_Notifications.NotificationCounter, {
    onClick: handleSidebarToggle,
    opened: !hidden
  }, notificationSummary), _react["default"].createElement(_Logo["default"], {
    link: homeLink,
    logo: logo
  }), _react["default"].createElement(_EzNavigation.MenuContent, {
    opened: !hidden
  }, _react["default"].createElement(_EzNavigation.Menus, {
    primary: true,
    "aria-label": t('Primary navigation')
  }, links.map(function (link, i) {
    return _react["default"].createElement(_Menu["default"], {
      key: i,
      link: link,
      sidebarToggle: handleSidebarToggle
    });
  })), utilityLinks && _react["default"].createElement(_EzNavigation.Menus, {
    "aria-label": t('Utility navigation')
  }, utilityLinks && utilityLinks.map(function (link, i) {
    return _react["default"].createElement(_Menu["default"], {
      key: i,
      link: link,
      sidebarToggle: handleSidebarToggle
    });
  })), userMenu && _react["default"].createElement(_EzNavigation.Menus, {
    "aria-label": t('User menu')
  }, _react["default"].createElement(_UserMenu["default"], {
    name: userMenu.name,
    links: userMenu.links,
    sidebarToggle: handleSidebarToggle
  })))), _react["default"].createElement(_EzNavigation.ContentContainer, {
    opened: hidden
  }, children));
};

var _default = EzNavigation;
exports["default"] = _default;
