// @ezcater/recipe/lib/components/EzNavigation/EzNavigation.styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ContentContainer = exports.Menus = exports.MenuContent = exports.NavWrapper = exports.Wrapper = void 0;

var _core = require("@emotion/core");

var _styled = _interopRequireDefault(require("../../themes/styled"));

var Wrapper =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "ee6zxjb0"
})("display:flex;flex:1 1 auto;flex-direction:column;height:100%;@media screen and (min-width:", function (props) {
  return props.theme.breakpoints.large;
}, "){flex-direction:row;}");
exports.Wrapper = Wrapper;
var NavWrapper =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "ee6zxjb1"
})("position:relative;background:linear-gradient(60deg,#1b2023 0%,#373d43 100%);@media screen and (min-width:", function (props) {
  return props.theme.breakpoints.large;
}, "){position:sticky;height:100vh;top:0;}[aria-label='Menu']{display:block;position:absolute;right:0;top:10px;@media screen and (min-width:", function (props) {
  return props.theme.breakpoints.large;
}, "){display:none;}}");
exports.NavWrapper = NavWrapper;
var MenuContent =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "ee6zxjb2"
})("visibility:hidden;height:0;transition:height 0.24s cubic-bezier(0.64,0,0.35,1);display:flex;flex-direction:column;overflow:hidden;", function (_ref) {
  var opened = _ref.opened;
  return opened &&
  /*#__PURE__*/
  (0, _core.css)("visibility:visible;height:calc(100vh - 70px);overflow-y:auto;");
}, " @media screen and (min-width:", function (props) {
  return props.theme.breakpoints.large;
}, "){visibility:visible;width:240px;transition:none;height:calc(100vh - 70px);overflow-y:auto;}");
exports.MenuContent = MenuContent;
var Menus =
/*#__PURE__*/
(0, _styled["default"])("nav", {
  target: "ee6zxjb3"
})("width:100%;", function (_ref2) {
  var primary = _ref2.primary;
  return primary &&
  /*#__PURE__*/
  (0, _core.css)("flex:1 1 0%;");
});
exports.Menus = Menus;
var ContentContainer =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "ee6zxjb4"
})("display:flex;flex-direction:column;flex:1 1 100%;min-width:0;max-width:100%;", function (_ref3) {
  var opened = _ref3.opened;
  return !opened &&
  /*#__PURE__*/
  (0, _core.css)("height:0;overflow:hidden;");
}, " @media screen and (min-width:", function (props) {
  return props.theme.breakpoints.large;
}, "){height:auto;overflow:visible;}");
exports.ContentContainer = ContentContainer;
