// @ezcater/recipe/lib/components/EzNavigation/Hamburger.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _core = require("@emotion/core");

var _styled = _interopRequireDefault(require("../../themes/styled"));

var HamburgerBox =
/*#__PURE__*/
(0, _styled["default"])("span", {
  target: "e1w0lti30"
})("position:relative;display:inline-block;width:40px;height:24px;");

var active = function active(_ref) {
  var isActive = _ref.active;
  return isActive &&
  /*#__PURE__*/
  (0, _core.css)("&,&:after,&:before{background-color:#fff;}transform:translate3d(0,10px,0) rotate(135deg);transition-delay:0.075s;&::before{transition-delay:0s;opacity:0;}&::after{transform:translate3d(0,-20px,0) rotate(-270deg);transition-delay:0.075s;}");
};

var HamburgerInner =
/*#__PURE__*/
(0, _styled["default"])("span", {
  target: "e1w0lti31"
})("display:block;top:2px;margin-top:-2px;&,&::before,&::after{width:40px;height:4px;background-color:#fff;border-radius:4px;position:absolute;transition-property:transform;}&::before,&::after{content:'';display:block;transition-duration:0.15s;transition-timing-function:ease;}&::before{top:-10px;}&::after{bottom:-10px;}transition-duration:0.275s;transition-timing-function:cubic-bezier(0.68,-0.55,0.265,1.55);&::before{top:10px;transition:opacity 0.125s 0.275s ease;}&::after{top:20px;transition:transform 0.275s cubic-bezier(0.68,-0.55,0.265,1.55);}", active);

var normalizeButton = function normalizeButton() {
  return (
    /*#__PURE__*/
    (0, _core.css)("font:inherit;color:inherit;text-transform:none;background-color:transparent;border:0;margin:0;overflow:visible;")
  );
};

var ToggleButton =
/*#__PURE__*/
(0, _styled["default"])("button", {
  target: "e1w0lti32"
})("padding:15px;display:inline-block;cursor:pointer;transition-property:opacity,filter;transition-duration:0.15s;transition-timing-function:linear;", normalizeButton, ";&:hover{opacity:0.7;}");

var Hamburger = function Hamburger(_ref2) {
  var opened = _ref2.opened,
      onClick = _ref2.onClick;
  return _react["default"].createElement(ToggleButton, {
    opened: opened,
    onClick: onClick,
    "aria-label": "Menu"
  }, _react["default"].createElement(HamburgerBox, null, _react["default"].createElement(HamburgerInner, {
    active: opened
  })));
};

var _default = Hamburger;
exports["default"] = _default;
