// @ezcater/recipe/lib/components/EzNavigation/Menu.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.menuStyles = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _core = require("@emotion/core");

var _styled = _interopRequireDefault(require("../../themes/styled"));

var _EzLink = _interopRequireDefault(require("../EzLink"));

var _utils = require("../../utils");

var _Notifications = require("./Notifications");

var menuStyles = function menuStyles(_ref) {
  var theme = _ref.theme;
  return (
    /*#__PURE__*/
    (0, _core.css)("color:#b8bdc2;font-weight:normal;display:flex;align-items:center;padding:", theme.spacing.md, " ", theme.spacing.xl4, " ", theme.spacing.md, " ", theme.spacing.xl, ";width:100%;position:relative;:hover:enabled{color:white;text-decoration:none;}:active:enabled,:not([aria-disabled='true'])[aria-expanded='true']{box-shadow:inset 0px 0px 10px #000000;}")
  );
};

exports.menuStyles = menuStyles;
var MenuLink =
/*#__PURE__*/
(0, _styled["default"])(_EzLink["default"], {
  target: "e1254gl10"
})("&&{", function (p) {
  return menuStyles(p);
}, ";}&.active{color:white;background-color:", function (_ref2) {
  var theme = _ref2.theme;
  return theme.colors.grays[900];
}, ";}");

var Notification = function Notification(_ref3) {
  var value = _ref3.value;
  return value === '★' ? _react["default"].createElement(_Notifications.Marketing, null, "\u2605") : _react["default"].createElement(_Notifications.Counter, null, value);
};

var Menu = function Menu(_ref4) {
  var link = _ref4.link,
      sidebarToggle = _ref4.sidebarToggle;
  return _react["default"].createElement("div", null, _react["default"].createElement(MenuLink, (0, _extends2["default"])({}, link, {
    onClick: (0, _utils.wrapEvent)(link.onClick, sidebarToggle),
    className: link.active ? 'active' : undefined
  }, link.to ? {
    activeClassName: 'active'
  } : {}), link.label, Boolean(link.notifications) && _react["default"].createElement(Notification, {
    value: link.notifications
  })));
};

var _default = Menu;
exports["default"] = _default;
