// @ezcater/recipe/lib/components/EzTable/useExpandedClickTarget.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = require("react");

var interactiveElements = 'label, input, button, select, a, [aria-role]';

var isInteractive = function isInteractive(e, context) {
  var el = e.target;

  do {
    if (el.matches(interactiveElements)) return el;
    el = el.parentNode;
  } while (el !== null && el.nodeType === 1 && el !== context);

  return null;
};

var useExpandedClickTarget = function useExpandedClickTarget() {
  var targetRef = (0, _react.useRef)();
  var clickableRef = (0, _react.useRef)();

  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      clickable = _useState2[0],
      setClickable = _useState2[1];

  return [targetRef, {
    onClick: function onClick(e) {
      if (targetRef.current && clickableRef.current && !isInteractive(e, clickableRef.current)) targetRef.current.click();
    },
    onMouseEnter: function onMouseEnter() {
      setClickable(Boolean(targetRef.current));
    },
    clickable: clickable,
    ref: clickableRef
  }];
};

var _default = useExpandedClickTarget;
exports["default"] = _default;
