// @ezcater/recipe/lib/styles/darken.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

/**
 * Converts a color from CSS hex format to CSS rgb format.
 */
function convertHexToRGB(color) {
  // eslint-disable-next-line no-param-reassign
  color = color.substr(1);
  var re = new RegExp(".{1,".concat(color.length / 3, "}"), 'g');
  var colors = color.match(re);
  return "rgb(".concat(colors.map(function (n) {
    return parseInt(n, 16);
  }).join(', '), ")");
}

/**
 * Returns an object with the type and values of a color.
 */
function decomposeColor(color) {
  if (color.charAt(0) === '#') return decomposeColor(convertHexToRGB(color));
  var marker = color.indexOf('(');
  var type = color.substring(0, marker);
  var values = color.substring(marker + 1, color.length - 1).split(',').map(function (value) {
    return parseFloat(value);
  });
  return {
    type: type,
    values: values
  };
}

function floatToHex(n) {
  var hex = parseInt(n, 10).toString(16);
  return hex.length === 1 ? "0".concat(hex) : hex;
}
/**
 * Converts a color object with type and values to a string.
 */


function recomposeColor(_ref) {
  var values = _ref.values;
  return "#".concat(values.map(floatToHex).join(''));
}

var clamp = function clamp(n) {
  var min = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 0;
  var max = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 1;
  return n < min ? min : n > max ? max : n;
};

function darken(hexColor, value) {
  var _decomposeColor = decomposeColor(hexColor),
      type = _decomposeColor.type,
      values = _decomposeColor.values;

  for (var i = 0; i < 3; i += 1) {
    values[i] *= 1 - clamp(value);
  }

  return recomposeColor({
    type: type,
    values: values
  });
}

var _default = darken;
exports["default"] = _default;
