// @ezcater/recipe/lib/components/EzInlineFeedback/EzInlineFeedback.styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InlineFeedback = void 0;

var _core = require("@emotion/core");

var _styledComponentVariant = _interopRequireDefault(require("styled-component-variant"));

var _styled = _interopRequireDefault(require("../../themes/styled"));

var base = function base(_ref) {
  var theme = _ref.theme;
  return (
    /*#__PURE__*/
    (0, _core.css)("display:inline-block;margin-left:", theme.spacing.sm, ";line-height:", theme.lineHeights.heading, ";vertical-align:top;svg{margin-right:", theme.spacing.xs, ";}")
  );
};

var useStyles = (0, _styledComponentVariant["default"])('use', {
  error: function error(_ref2) {
    var theme = _ref2.theme;
    return (
      /*#__PURE__*/
      (0, _core.css)("color:", theme.colors.destructive.main, ";svg path{fill:", theme.colors.destructive.main, ";}")
    );
  },
  progress: function progress(_ref3) {
    var theme = _ref3.theme;
    return (
      /*#__PURE__*/
      (0, _core.css)("svg path{fill:", theme.colors.text.deemphasis, ";}")
    );
  },
  success: function success(_ref4) {
    var theme = _ref4.theme;
    return (
      /*#__PURE__*/
      (0, _core.css)("svg path{fill:", theme.colors.success.text, ";}")
    );
  }
});
var InlineFeedback =
/*#__PURE__*/
(0, _styled["default"])("span", {
  target: "e1f9xte90"
})(base, useStyles);
exports.InlineFeedback = InlineFeedback;
