// @ezcater/recipe/lib/components/EzPageHeader/Tabs.styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledLi = exports.StyledUl = void 0;

var _core = require("@emotion/core");

var _styled = _interopRequireDefault(require("../../themes/styled"));

var _styles = require("../../styles");

var listBase = function listBase(_ref) {
  var theme = _ref.theme;
  return (
    /*#__PURE__*/
    (0, _core.css)("display:flex;margin:0;padding:0;list-style:none;&& > * + *{margin-left:", theme.spacing.sm, ";}")
  );
};

var activeTab = function activeTab(_ref2) {
  var theme = _ref2.theme,
      active = _ref2.active;
  return active &&
  /*#__PURE__*/
  (0, _core.css)("box-shadow:inset 0 -2px 0 0 ", theme.colors.interactive.focus.outline, ";&& > *{color:", theme.colors.text.base, ";:focus{outline-color:", theme.colors.interactive.focus.outline, ";}}");
};

var tabBase = function tabBase(_ref3) {
  var theme = _ref3.theme,
      active = _ref3.active;
  return (
    /*#__PURE__*/
    (0, _core.css)("display:flex;margin:0;&& a,&& > button{display:flex;padding:", theme.spacing.sm, " ", theme.spacing.xs, ";text-decoration:none;&:hover{box-shadow:", !active && "inset 0 -2px 0 0 ".concat((0, _styles.darken)(theme.colors.border.base, 0.15)), ";}&:focus{outline:auto;outline-color:", theme.colors.interactive.focus.outline, ";}}")
  );
};

var StyledUl =
/*#__PURE__*/
(0, _styled["default"])("ul", {
  target: "e1umt9pu0"
})(listBase);
exports.StyledUl = StyledUl;
var StyledLi =
/*#__PURE__*/
(0, _styled["default"])("li", {
  target: "e1umt9pu1"
})(tabBase, activeTab);
exports.StyledLi = StyledLi;
