// @ezcater/recipe/lib/utils/hooks/usePopper.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _core = require("@popperjs/core");

var _react = require("react");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var _default = function _default() {
  var options = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var referenceRef = (0, _react.useRef)();
  var popperRef = (0, _react.useRef)();
  var instanceRef = (0, _react.useRef)();
  var combinedOptions = (0, _react.useMemo)(function () {
    var _options$modifiers = options.modifiers,
        modifiers = _options$modifiers === void 0 ? [] : _options$modifiers;
    return _objectSpread({}, options, {
      modifiers: [].concat((0, _toConsumableArray2["default"])(modifiers), [{
        name: 'preventOverflow',
        options: {
          padding: 8
        }
      }])
    });
  }, [options]);
  (0, _react.useLayoutEffect)(function () {
    var instance = (0, _core.createPopper)(referenceRef.current, popperRef.current, combinedOptions);
    instanceRef.current = instance;
    return function () {
      instance.destroy();
    };
  }, [combinedOptions]);
  (0, _react.useLayoutEffect)(function () {
    instanceRef.current.setOptions(combinedOptions);
    instanceRef.current.update();
  }, [combinedOptions]);
  return {
    reference: referenceRef,
    popper: popperRef,
    instance: instanceRef
  };
};

exports["default"] = _default;
