// @ezcater/recipe/lib/components/EzModal/EzModal.styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ButtonFooter = exports.ContentContainer = exports.HeaderContainer = exports.StyledDialog = exports.StyledOverlay = void 0;

var _core = require("@emotion/core");

var _styled = _interopRequireDefault(require("../../themes/styled"));

var StyledOverlay =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "e1ozo9wc0"
})("align-items:center;background-color:rgba(0,0,0,0.6);display:flex;flex-direction:column;justify-content:center;overflow:hidden;position:fixed;top:0;right:0;bottom:0;left:0;");
exports.StyledOverlay = StyledOverlay;
var StyledDialog =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "e1ozo9wc1"
})("background:", function (props) {
  return props.theme.colors.white;
}, ";display:flex;flex-direction:column;height:100%;max-height:100vh;outline:none;width:100%;overflow-y:auto;@media screen and (min-width:", function (props) {
  return props.theme.breakpoints.medium;
}, "){border-radius:12px;height:auto;max-height:calc(100vh - ", function (props) {
  return props.theme.spacing.xl4;
}, ");width:575px;}:focus{box-shadow:0px 0px 2px 2px ", function (_ref) {
  var theme = _ref.theme;
  return theme.colors.interactive.focus.outline;
}, ",0 1px 1px 0 rgba(0,0,0,0.12);}");
exports.StyledDialog = StyledDialog;

var containerPadding = function containerPadding(_ref2) {
  var _ref2$theme = _ref2.theme,
      spacing = _ref2$theme.spacing,
      breakpoints = _ref2$theme.breakpoints;
  return (
    /*#__PURE__*/
    (0, _core.css)("padding:", spacing.md, " ", spacing.xl, ";@media screen and (min-width:", breakpoints.medium, "){padding:", spacing.xl, " ", spacing.xl2, ";}")
  );
};

var HeaderContainer =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "e1ozo9wc2"
})(function (props) {
  return containerPadding({
    theme: props.theme
  });
}, " border-bottom:1px solid ", function (props) {
  return props.theme.colors.border.subtle;
}, ";display:flex;flex:none;justify-content:space-between;@media screen and (min-width:", function (props) {
  return props.theme.breakpoints.medium;
}, "){border-bottom:none;}button svg path{fill:#3a81be;}");
exports.HeaderContainer = HeaderContainer;
var ContentContainer =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "e1ozo9wc3"
})(function (props) {
  return containerPadding({
    theme: props.theme
  });
}, " flex:auto;overflow-y:auto;&& > * + *{margin-top:", function (props) {
  return props.theme.spacing.lg;
}, ";}@media screen and (min-width:", function (props) {
  return props.theme.breakpoints.medium;
}, "){padding-top:0;}");
exports.ContentContainer = ContentContainer;
var ButtonFooter =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "e1ozo9wc4"
})(function (props) {
  return containerPadding({
    theme: props.theme
  });
}, " background:", function (props) {
  return props.theme.colors.page.background;
}, ";border-top:1px solid ", function (props) {
  return props.theme.colors.border.subtle;
}, ";flex:none;");
exports.ButtonFooter = ButtonFooter;
