// @ezcater/recipe/lib/components/EzModal/ScrollLock.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.preventScroll = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _reactEmotion = require("react-emotion");

var preventScroll =
/*#__PURE__*/
(0, _reactEmotion.css)("overflow:hidden;");
exports.preventScroll = preventScroll;

var disableScroll = function disableScroll() {
  return document && document.querySelector('html').classList.add(preventScroll);
};

var enableScroll = function enableScroll() {
  return document && document.querySelector('html').classList.remove(preventScroll);
};

var ScrollLock =
/*#__PURE__*/
function (_React$Component) {
  (0, _inherits2["default"])(ScrollLock, _React$Component);

  function ScrollLock() {
    (0, _classCallCheck2["default"])(this, ScrollLock);
    return (0, _possibleConstructorReturn2["default"])(this, (0, _getPrototypeOf2["default"])(ScrollLock).apply(this, arguments));
  }

  (0, _createClass2["default"])(ScrollLock, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      disableScroll();
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      enableScroll();
    }
  }, {
    key: "render",
    value: function render() {
      return null;
    }
  }]);
  return ScrollLock;
}(_react["default"].Component);

exports["default"] = ScrollLock;
