// @ezcater/recipe/lib/components/CloseButton/CloseButton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _CloseButton = require("./CloseButton.styles");

var _styled = _interopRequireDefault(require("../../themes/styled"));

var CloseButton =
/*#__PURE__*/
(0, _styled["default"])("button", {
  target: "e1iconj00"
})(_CloseButton.base);

var _default = function _default(_ref) {
  var label = _ref.label,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["label"]);
  return _react["default"].createElement(CloseButton, (0, _extends2["default"])({}, props, {
    "aria-label": label
  }), _react["default"].createElement("svg", {
    xmlns: "http://www.w3.org/2000/svg",
    width: "16",
    height: "16",
    version: "1"
  }, _react["default"].createElement("path", {
    fill: "currentColor",
    fillRule: "evenodd",
    d: "M8 6l5-5 2 2-5 5 5 5-2 2-5-5-5 5-2-2 5-5-5-5 2-2 5 5z"
  })));
};

exports["default"] = _default;
