// @ezcater/recipe/lib/components/EzToggle/EzToggle.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _EzToggle = require("./EzToggle.styles");

var _EzLabel = _interopRequireDefault(require("../EzLabel"));

var _EzInlineFeedback = _interopRequireDefault(require("../EzInlineFeedback"));

var _hooks = require("../../utils/hooks");

var EzToggle = (0, _react.forwardRef)(function (_ref, ref) {
  var id = _ref.id,
      label = _ref.label,
      status = _ref.status,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["id", "label", "status"]);

  var _useState = (0, _react.useState)(props.checked),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      checked = _useState2[0],
      setChecked = _useState2[1];

  var uniqueId = (0, _hooks.useUniqueId)();
  var idOrDefault = id || uniqueId;
  var inputRef = (0, _react.useRef)(null);

  var handleClick = function handleClick(event) {
    var checkbox = inputRef.current;

    if (event.target !== checkbox) {
      event.preventDefault();
      checkbox.focus();
      checkbox.click();
      return;
    }

    setChecked('checked' in props ? props.checked : checkbox.checked);
  };

  return _react["default"].createElement(_EzToggle.Container, {
    disabled: props.disabled,
    checked: checked,
    onClick: handleClick,
    ref: ref
  }, _react["default"].createElement("input", (0, _extends2["default"])({}, props, {
    id: idOrDefault,
    ref: inputRef,
    type: "checkbox"
  })), _react["default"].createElement(_EzToggle.Track, null), label && _react["default"].createElement(_EzLabel["default"], {
    htmlFor: idOrDefault,
    position: "right",
    as: "label"
  }, label), status && _react["default"].createElement(_EzInlineFeedback["default"], {
    use: status
  }));
});
var _default = EzToggle;
exports["default"] = _default;
