// @ezcater/recipe/lib/components/EzTooltip/EzTooltip.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _EzTooltip = require("./EzTooltip.styles");

var _hooks = require("../../utils/hooks");

var _EzPopover = _interopRequireDefault(require("../EzPopover"));

var EzTooltip = function EzTooltip(_ref) {
  var children = _ref.children,
      message = _ref.message,
      position = _ref.position,
      rest = (0, _objectWithoutProperties2["default"])(_ref, ["children", "message", "position"]);
  var id = (0, _hooks.useUniqueId)();
  var targetRef = (0, _react.useRef)(null);

  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      isHovered = _useState2[0],
      setIsHovered = _useState2[1];

  var _useState3 = (0, _react.useState)(false),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      isFocused = _useState4[0],
      setIsFocused = _useState4[1];

  var onMouseEnterEvent = function onMouseEnterEvent() {
    return setIsHovered(true);
  };

  var onMouseLeaveEvent = function onMouseLeaveEvent() {
    return setIsHovered(false);
  };

  var onFocusEvent = function onFocusEvent() {
    return setIsFocused(true);
  };

  var onBlurEvent = function onBlurEvent() {
    return setIsFocused(false);
  };

  var showTooltip = isHovered || isFocused;

  var child = _react["default"].Children.only(children);

  var childProps = {
    'aria-describedby': id,
    onMouseEnter: onMouseEnterEvent,
    onMouseLeave: onMouseLeaveEvent,
    onFocus: onFocusEvent,
    onBlur: onBlurEvent,
    ref: targetRef
  };
  return _react["default"].createElement(_react["default"].Fragment, null, _react["default"].cloneElement(child, childProps), showTooltip && _react["default"].createElement(_EzPopover["default"], {
    targetRef: targetRef,
    placement: position === 'horizontal' ? 'right' : 'bottom',
    modifiers: [{
      name: 'arrow',
      enabled: true,
      options: {
        padding: 5
      }
    }, {
      name: 'offset',
      options: {
        offset: [0, 5]
      }
    }]
  }, _react["default"].createElement(_EzTooltip.TooltipArrow, {
    "data-popper-arrow": true
  }, _react["default"].createElement("svg", {
    width: "10",
    height: "10"
  }, _react["default"].createElement("path", {
    d: "M0 5l5-5 5 5z"
  }), _react["default"].createElement("path", {
    d: "M1.5 5L5 1.5 8.5 5z"
  }))), _react["default"].createElement(_EzTooltip.Tooltip, (0, _extends2["default"])({
    role: "tooltip",
    id: id,
    tabIndex: "-1"
  }, rest), message && _react["default"].createElement(_EzTooltip.Message, null, message))));
};

EzTooltip.defaultProps = {
  position: 'vertical'
};
var _default = EzTooltip;
exports["default"] = _default;
