// @ezcater/recipe/lib/components/EzTooltip/EzTooltip.styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TooltipArrow = exports.Tooltip = exports.Message = void 0;

var _core = require("@emotion/core");

var _styled = _interopRequireDefault(require("../../themes/styled"));

var base = function base() {
  return (
    /*#__PURE__*/
    (0, _core.css)("position:relative;padding:0.5rem;max-width:300px;box-shadow:0 1px 1px 0 rgba(0,0,0,0.12);white-space:pre-line;")
  );
};

var color = function color(_ref) {
  var theme = _ref.theme;
  var backgroundColor = theme.colors.grays['800'];
  var textColor = theme.colors.white;
  return (
    /*#__PURE__*/
    (0, _core.css)("color:", textColor, ";border:1px solid ", backgroundColor, ";background-color:", backgroundColor, ";")
  );
};

var Message =
/*#__PURE__*/
(0, _styled["default"])("p", {
  target: "er7tgdd0"
})("margin-top:0;margin-bottom:0;");
exports.Message = Message;
var Tooltip =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "er7tgdd1"
})(base, color);
exports.Tooltip = Tooltip;
var TooltipArrow =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "er7tgdd2"
})("> svg{position:absolute;pointer-events:none;fill:", function (_ref2) {
  var theme = _ref2.theme;
  return theme.colors.grays['800'];
}, ";}[data-popper-placement^='top'] > &[data-popper-arrow]{bottom:5px;svg{transform:rotate(180deg) translate(50%,0);}}[data-popper-placement^='right'] > &[data-popper-arrow]{left:-5px;svg{transform:rotate(-90deg) translate(50%,0);}}[data-popper-placement^='bottom'] > &[data-popper-arrow]{top:-5px;svg{transform:translate(-50%,0);}}[data-popper-placement^='left'] > &[data-popper-arrow]{right:5px;svg{transform:rotate(90deg) translate(-50%,0);}}");
exports.TooltipArrow = TooltipArrow;
