// @ezcater/recipe/lib/components/EzStatus/EzStatus.styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EzStatusContainer = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _core = require("@emotion/core");

var _styledComponentVariant = _interopRequireDefault(require("styled-component-variant"));

var _styled = _interopRequireDefault(require("../../themes/styled"));

var coloredStyles = function coloredStyles(getColors) {
  return function (_ref) {
    var colors = _ref.theme.colors;

    var _getColors = getColors(colors),
        _getColors2 = (0, _slicedToArray2["default"])(_getColors, 2),
        lightColor = _getColors2[0],
        darkColor = _getColors2[1];

    return (
      /*#__PURE__*/
      (0, _core.css)("background-color:", lightColor, ";color:", darkColor, ";")
    );
  };
};

var use = (0, _styledComponentVariant["default"])('use', {
  neutral: coloredStyles(function (colors) {
    return [colors.page.background, colors.text.base];
  }),
  success: coloredStyles(function (colors) {
    return [colors.success.background, colors.success.text];
  }),
  informational: coloredStyles(function (colors) {
    return [colors.info.background, colors.info.text];
  }),
  attention: coloredStyles(function (colors) {
    return [colors.info.background, colors.info.text];
  }),
  warning: coloredStyles(function (colors) {
    return [colors.warning.background, colors.warning.text];
  }),
  error: coloredStyles(function (colors) {
    return [colors.destructive.background, colors.destructive.text];
  })
});

var base = function base(_ref2) {
  var theme = _ref2.theme,
      size = _ref2.size;
  return (
    /*#__PURE__*/
    (0, _core.css)("padding:", theme.spacing.xs2, " ", theme.spacing.sm, ";line-height:1;font-size:", size === 'small' ? theme.fontSizes[200] : theme.fontSizes[400], ";font-weight:bold;border-radius:1em;")
  );
};

var EzStatusContainer =
/*#__PURE__*/
(0, _styled["default"])("span", {
  target: "e7n54010"
})(base, use);
exports.EzStatusContainer = EzStatusContainer;
