// @ezcater/recipe/lib/components/EzCard/EzCard.styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SectionContainer = exports.CardLayout = exports.CardHeadingContainer = exports.CardContainer = void 0;

var _core = require("@emotion/core");

var _styled = _interopRequireDefault(require("../../themes/styled"));

var accentStyles = function accentStyles(_ref) {
  var accent = _ref.accent,
      theme = _ref.theme;
  return accent &&
  /*#__PURE__*/
  (0, _core.css)("border-left:5px solid ", theme.colors.info.foreground, ";");
};

var CardContainer =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "ekvwvyw0"
})("background:", function (props) {
  return props.theme.colors.white;
}, ";border:1px solid ", function (props) {
  return props.theme.colors.border.base;
}, ";border-radius:", function (props) {
  return props.theme.borderRadius[2];
}, ";", accentStyles, ";");
exports.CardContainer = CardContainer;
var CardHeadingContainer =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "ekvwvyw1"
})("margin:", function (props) {
  return props.theme.spacing.lg;
}, " ", function (props) {
  return props.theme.spacing.lg;
}, " 0;");
exports.CardHeadingContainer = CardHeadingContainer;

var gutter = function gutter(_ref2) {
  var theme = _ref2.theme;
  return (
    /*#__PURE__*/
    (0, _core.css)("margin:calc(", theme.spacing.sm, " * -1) 0 0 calc(", theme.spacing.sm, " * -1);> *{margin:", theme.spacing.sm, " 0 0 ", theme.spacing.sm, ";}")
  );
};

var CardLayout =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "ekvwvyw2"
})("display:flex;flex-wrap:wrap;align-items:baseline;justify-content:space-between;", gutter, ";");
exports.CardLayout = CardLayout;

var vertical = function vertical(_ref3) {
  var horizontal = _ref3.horizontal,
      theme = _ref3.theme;
  return !horizontal &&
  /*#__PURE__*/
  (0, _core.css)("&& > * + *{border-top:1px solid ", theme.colors.border.base, ";}");
};

var horizontal = function horizontal(_ref4) {
  var isHorizontal = _ref4.horizontal;
  return isHorizontal &&
  /*#__PURE__*/
  (0, _core.css)("display:flex;> *{flex-basis:0;flex-grow:1;}");
};

var SectionContainer =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "ekvwvyw3"
})(vertical, horizontal);
exports.SectionContainer = SectionContainer;
