// @ezcater/recipe/lib/components/EzCard/EzCard.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _EzCard = require("./EzCard.styles");

var _EzLink = _interopRequireDefault(require("../EzLink"));

var _EzCardFooter = _interopRequireDefault(require("./EzCardFooter"));

var _EzCardSection = _interopRequireDefault(require("./EzCardSection"));

var _utils = require("../../utils");

var _EzHeading = _interopRequireDefault(require("../EzHeading"));

function isEzCardSection(element) {
  var _element$type;

  return (element === null || element === void 0 ? void 0 : (_element$type = element.type) === null || _element$type === void 0 ? void 0 : _element$type.displayName) === 'EzCardSection';
}

function wrappedChildren(children) {
  if (isEzCardSection((children === null || children === void 0 ? void 0 : children[0]) || children)) return children;
  return _react["default"].createElement(_EzCardSection["default"], null, children);
}

var isExpandableCardFooter = function isExpandableCardFooter(expandable) {
  return expandable && expandable.expandLabel !== undefined;
};
/**
 * Cards are the primary means of grouping sections on a page.
 */


var EzCard = function EzCard(_ref) {
  var title = _ref.title,
      subtitle = _ref.subtitle,
      accent = _ref.accent,
      actions = _ref.actions,
      expandable = _ref.expandable,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["title", "subtitle", "accent", "actions", "expandable"]);

  var heading = title && _react["default"].createElement(_EzHeading["default"], {
    size: "3",
    subheading: subtitle
  }, title);

  var layout = actions && _react["default"].createElement(_EzCard.CardLayout, null, _react["default"].createElement("div", null, heading), _react["default"].createElement("div", null, actions));

  var horizontal = props.horizontal,
      children = props.children;
  return _react["default"].createElement(_EzCard.CardContainer, (0, _extends2["default"])({}, (0, _utils.filterValidProps)(props), {
    accent: accent
  }), title && _react["default"].createElement(_EzCard.CardHeadingContainer, null, actions ? layout : heading), _react["default"].createElement(_EzCard.SectionContainer, {
    horizontal: horizontal
  }, wrappedChildren(children)), isExpandableCardFooter(expandable) && _react["default"].createElement(_EzCardFooter["default"], null, _react["default"].createElement(_EzLink["default"], {
    href: "#",
    onClick: expandable.onClick
  }, expandable.isExpanded && expandable.collapseLabel ? expandable.collapseLabel : expandable.expandLabel)));
};

EzCard.displayName = 'EzCard';
/**
 * @component
 */

var _default = EzCard;
exports["default"] = _default;
