// @ezcater/recipe/lib/components/EzSuperRadioButtons/EzSuperRadioButtons.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireWildcard(require("react"));

var _hooks = require("../../utils/hooks");

var _EzSuperRadioButtons = require("./EzSuperRadioButtons.styles");

var getId = function getId(name, _ref) {
  var value = _ref.value;
  return "".concat(name, "-").concat(value);
};

var EzSuperRadioButtons = (0, _react.forwardRef)(function (_ref2, ref) {
  var options = _ref2.options,
      label = _ref2.label,
      value = _ref2.value,
      _onChange = _ref2.onChange;
  var id = (0, _hooks.useUniqueId)();
  return _react["default"].createElement(_EzSuperRadioButtons.EzSuperRadioButtonsGroup, {
    ref: ref,
    role: "radiogroup",
    "aria-labelledby": "radiogroup-".concat(id)
  }, _react["default"].createElement(_EzSuperRadioButtons.EzSuperRadioButtonsLabel, {
    id: "radiogroup-".concat(id)
  }, label), options.map(function (option) {
    return _react["default"].createElement(_react.Fragment, {
      key: option.value
    }, _react["default"].createElement("input", (0, _extends2["default"])({
      id: getId(id, option),
      type: "radio",
      name: id,
      value: option.value
    }, _onChange ? {
      checked: option.value === value
    } : {
      defaultChecked: option.value === value
    }, {
      onChange: function onChange(e) {
        return _onChange && _onChange(e.target.value);
      },
      disabled: option.disabled
    })), _react["default"].createElement(_EzSuperRadioButtons.EzSuperRadioButtonsItem, {
      key: option.value,
      htmlFor: getId(id, option)
    }, _react["default"].createElement(_EzSuperRadioButtons.RadioButton, {
      width: "24",
      height: "24",
      checked: option.value === value
    }), option.imageSrc && _react["default"].createElement(_EzSuperRadioButtons.EzSuperRadioButtonsImageWrapper, null, _react["default"].createElement(_EzSuperRadioButtons.EzSuperRadioButtonsImage, {
      src: option.imageSrc,
      alt: ""
    })), _react["default"].createElement(_EzSuperRadioButtons.EzSuperRadioButtonsItemLabel, null, option.label)));
  }));
});
EzSuperRadioButtons.displayName = 'EzSuperRadioButtons';
var _default = EzSuperRadioButtons;
exports["default"] = _default;
