// @ezcater/recipe/lib/components/EzLayout/EzLayout.styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.spacing = exports.layout = exports.stack = exports.split = exports.equal = exports.right = exports.basic = exports.base = void 0;

var _core = require("@emotion/core");

var _styles = require("../../styles");

var base = function base() {
  return (
    /*#__PURE__*/
    (0, _core.css)("display:flex;")
  );
};

exports.base = base;

var reset = function reset() {
  return (
    /*#__PURE__*/
    (0, _core.css)("align-items:normal;flex-direction:row;justify-content:normal;> *{flex-basis:auto;flex-grow:0;}")
  );
};

var basic = function basic() {
  return (
    /*#__PURE__*/
    (0, _core.css)("align-items:center;justify-content:flex-start;")
  );
};

exports.basic = basic;

var right = function right() {
  return (
    /*#__PURE__*/
    (0, _core.css)("align-items:center;justify-content:flex-end;")
  );
};

exports.right = right;

var equal = function equal() {
  return (
    /*#__PURE__*/
    (0, _core.css)("align-items:center;justify-content:space-between;> *{flex-basis:0;flex-grow:1;}")
  );
};

exports.equal = equal;

var split = function split() {
  return (
    /*#__PURE__*/
    (0, _core.css)("align-items:center;justify-content:space-between;")
  );
};

exports.split = split;

var stack = function stack() {
  return (
    /*#__PURE__*/
    (0, _core.css)("flex-direction:column;")
  );
};

exports.stack = stack;
var layout = (0, _styles.responsive)('layout', {
  basic: basic,
  right: right,
  equal: equal,
  split: split,
  stack: stack,
  reset: reset
});
exports.layout = layout;

var setSpacing = function setSpacing(top, left) {
  return (
    /*#__PURE__*/
    (0, _core.css)("&& > * + *{margin-top:", top, ";margin-left:", left, ";}")
  );
};

var leftSpacing = function leftSpacing(_ref) {
  var theme = _ref.theme;
  return setSpacing(null, theme.spacing.sm);
};

var topSpacing = function topSpacing(_ref2) {
  var theme = _ref2.theme;
  return setSpacing(theme.spacing.sm, null);
};

var spacing = (0, _styles.responsive)('layout', {
  basic: leftSpacing,
  right: leftSpacing,
  equal: leftSpacing,
  stack: topSpacing,
  reset: function reset() {
    return setSpacing(0, 0);
  }
});
exports.spacing = spacing;
