// @ezcater/recipe/lib/components/EzSegmentedControl/EzSegmentedControl.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireWildcard(require("react"));

var _EzSegmentedControl = require("./EzSegmentedControl.styles");

var _EzLabel = _interopRequireDefault(require("../EzLabel"));

var getId = function getId(name, _ref) {
  var value = _ref.value;
  return "".concat(name, "-").concat(value);
};

/**
 * EzSegmentedControls present mutually exclusive options as buttons.
 */
var EzSegmentedControl = function EzSegmentedControl(_ref2) {
  var active = _ref2.active,
      className = _ref2.className,
      label = _ref2.label,
      labelPosition = _ref2.labelPosition,
      name = _ref2.name,
      _onChange = _ref2.onChange,
      options = _ref2.options;
  return _react["default"].createElement(_EzSegmentedControl.Fieldset, {
    className: className,
    role: "radiogroup",
    "aria-labelledby": "radiogroup-".concat(name)
  }, _react["default"].createElement(_EzLabel["default"], {
    id: "radiogroup-".concat(name),
    position: labelPosition
  }, label), options.map(function (option) {
    return _react["default"].createElement(_react.Fragment, {
      key: option.value
    }, _react["default"].createElement("input", (0, _extends2["default"])({
      id: getId(name, option),
      type: "radio",
      name: name
    }, _onChange ? {
      checked: option.value === active
    } : {
      defaultChecked: option.value === active
    }, {
      onChange: function onChange() {
        return _onChange && _onChange(option.value);
      },
      disabled: option.disabled
    })), _react["default"].createElement(_EzSegmentedControl.Option, {
      key: option.value,
      htmlFor: getId(name, option)
    }, option.label));
  }));
};
/**
 * defaultProps
 * @property {bool} active - No option is selected by default.
 * @property {string} labelPosition - The label is positioned on the left of the control by default.
 */


EzSegmentedControl.defaultProps = {
  labelPosition: 'left'
};
/**
 * @component
 */

var _default = EzSegmentedControl;
exports["default"] = _default;
