// @ezcater/recipe/lib/components/EzField/EzChoice.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _EzCheckbox = _interopRequireDefault(require("../EzCheckbox"));

var _styled = _interopRequireDefault(require("../../themes/styled"));

var _EzRadioButton = _interopRequireDefault(require("../EzRadioButton"));

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var Options =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "evekzw00"
})("> label + label{margin-top:", function (_ref) {
  var theme = _ref.theme;
  return theme.spacing.xs;
}, ";}");
var Label =
/*#__PURE__*/
(0, _styled["default"])("label", {
  target: "evekzw01"
})("display:block;margin-top:", function (_ref2) {
  var theme = _ref2.theme;
  return theme.spacing.xs;
}, ";> *{margin-right:", function (_ref3) {
  var theme = _ref3.theme;
  return theme.spacing.xs;
}, ";}");

var _default = function _default(props) {
  var type = props.type,
      fieldName = props.name,
      _props$value = props.value,
      selected = _props$value === void 0 ? [] : _props$value,
      options = props.options,
      onChange = props.onChange,
      onFocus = props.onFocus,
      onBlur = props.onBlur;
  var multiple = type === 'checkbox';
  var name = multiple ? "".concat(fieldName, "[]") : fieldName;

  function handleChange(e) {
    var _e$currentTarget = e.currentTarget,
        value = _e$currentTarget.value,
        checked = _e$currentTarget.checked;
    var newSelection = multiple ? checked ? [value].concat(selected) : selected.filter(function (s) {
      return s !== value;
    }) : value;
    onChange(_objectSpread({}, e, {
      target: _objectSpread({}, props, {
        checked: checked,
        value: newSelection
      })
    }));
  }

  return _react["default"].createElement(Options, null, options.map(function (choice, i) {
    var label = choice.label,
        disabled = choice.disabled,
        value = choice.value;
    var inputProps = {
      checked: 'value' in props ? multiple ? selected.indexOf(value) >= 0 : value === selected : undefined,
      disabled: disabled,
      label: label,
      name: name,
      onChange: handleChange,
      onFocus: onFocus,
      onBlur: onBlur,
      type: type,
      value: value
    };

    var input = _react["default"].createElement(multiple ? _EzCheckbox["default"] : _EzRadioButton["default"], inputProps);

    return _react["default"].createElement(Label, {
      key: i
    }, input, label);
  }));
};

exports["default"] = _default;
