// @ezcater/recipe/lib/components/EzField/EzCombobox.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useComboboxFlyout = exports.useComboboxInput = exports.useCombobox = exports.useComboboxState = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = require("react");

var _Disclosure = require("reakit/Disclosure");

var _hooks = require("../../utils/hooks");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var useComboboxState = function useComboboxState() {
  // eslint-disable-next-line @typescript-eslint/camelcase
  var hidden = (0, _Disclosure.useDisclosureState)({
    unstable_isMounted: true
  });
  return _objectSpread({}, hidden);
};

exports.useComboboxState = useComboboxState;

var useCombobox = function useCombobox(options, htmlProps) {
  var ref = (0, _react.useRef)();
  var optionsRef = (0, _react.useRef)();
  (0, _hooks.useOnClickOutside)(options.hide, [ref, optionsRef]);
  (0, _hooks.useEventListenerOutside)(options.hide, 'focusin', [ref, optionsRef]);
  return _objectSpread({
    role: 'combobox',
    ref: ref,
    optionsRef: optionsRef,
    'aria-expanded': Boolean(options.visible),
    'aria-owns': options.unstable_hiddenId
  }, htmlProps);
};

exports.useCombobox = useCombobox;

var useComboboxInput = function useComboboxInput(options, htmlProps) {
  var ref = (0, _react.useRef)();

  var onKeyDown = function onKeyDown(e) {
    var keyMap = {
      Escape: options.hide,
      ArrowUp: !options.visible && options.show,
      ArrowDown: !options.visible && options.show,
      ' ': !options.visible && options.show
    };
    var action = keyMap[e.key];

    if (action) {
      e.preventDefault();
      action(e);
    }
  };

  return _objectSpread({
    type: 'text',
    ref: ref,
    'aria-controls': options.unstable_hiddenId
  }, htmlProps, {
    onMouseDown: (0, _hooks.useAllCallbacks)(options.toggle, htmlProps.onMouseDown),
    onKeyDown: (0, _hooks.useAllCallbacks)(onKeyDown, htmlProps.onKeyDown)
  });
};

exports.useComboboxInput = useComboboxInput;

var useComboboxFlyout = function useComboboxFlyout(options) {
  var htmlProps = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};
  (0, _Disclosure.useDisclosureContent)(options, htmlProps);
  return {
    id: options.unstable_hiddenId
  };
};

exports.useComboboxFlyout = useComboboxFlyout;
