// @ezcater/recipe/lib/components/EzField/EzCombobox.styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Combobox = exports.Container = void 0;

var _core = require("@emotion/core");

var _styled = _interopRequireDefault(require("../../themes/styled"));

var errorIconPosition = function errorIconPosition(_ref) {
  var hasError = _ref.hasError;
  return hasError &&
  /*#__PURE__*/
  (0, _core.css)("+ * > svg{right:1em;}");
};

var hideErrorMessageWhenOpen = function hideErrorMessageWhenOpen(_ref2) {
  var hasError = _ref2.hasError,
      opened = _ref2.opened;
  return hasError && opened &&
  /*#__PURE__*/
  (0, _core.css)("+ * + *{display:none;}");
};

var Container =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "eruyps80"
})("input{cursor:default;text-overflow:ellipsis;white-space:nowrap;overflow:hidden;}", errorIconPosition, ";", hideErrorMessageWhenOpen, ";");
exports.Container = Container;

var iconColor = function iconColor(p) {
  return p.disabled ? p.theme.colors.interactive.disabled.foreground : p.theme.colors.interactive.base;
};

var Combobox =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "eruyps81"
})("&& input{padding-right:2em;}svg{stroke:", iconColor, ";transition:transform 0.3s;}");
exports.Combobox = Combobox;
