// @ezcater/recipe/lib/components/EzField/EzField.styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ScreenReaderOnly = exports.ErrorTriangle = exports.CharacterLimit = exports.InlineError = exports.Helper = exports.Field = exports.CustomInputWrapper = void 0;

var _core = require("@emotion/core");

var _styles = require("../../styles");

var _styled = _interopRequireDefault(require("../../themes/styled"));

var _EzTextInput = _interopRequireDefault(require("./EzTextInput.styles"));

var _Icons = require("../Icons");

// pre-calculate where to put the error icon (icon width + right padding of input)
var calloutBorderRadius = '3px';
var roundedFull = {
  borderRadius: calloutBorderRadius
};
var roundedBottom =
/*#__PURE__*/
(0, _core.css)("border-bottom-right-radius:", calloutBorderRadius, ";border-bottom-left-radius:", calloutBorderRadius, ";");
var CustomInputWrapper =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "e3hq9p20"
})("> input,> textarea{", _EzTextInput["default"], ";}");
exports.CustomInputWrapper = CustomInputWrapper;
var Field =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "e3hq9p21"
})("position:relative;border:none;margin:0;padding:0;> * + *{margin-top:", function (_ref) {
  var theme = _ref.theme;
  return theme.spacing.xs;
}, ";}");
exports.Field = Field;
var Helper =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "e3hq9p22"
})("font-size:", function (_ref2) {
  var theme = _ref2.theme;
  return theme.fontSizes[200];
}, ";color:", function (_ref3) {
  var theme = _ref3.theme;
  return theme.colors.text.deemphasis;
}, ";margin-top:", function (_ref4) {
  var theme = _ref4.theme;
  return theme.spacing.xs;
}, ";");
exports.Helper = Helper;
var detached =
/*#__PURE__*/
(0, _core.css)("box-shadow:0 2px 4px 0 rgba(34,36,38,0.12),0 2px 10px 0 rgba(34,36,38,0.15);", roundedFull, ";");
var inlineMessageOffset = {
  transform: 'translate3d(0, -2px, 0)'
};

var callout = function callout(_ref5) {
  var theme = _ref5.theme,
      active = _ref5.active,
      showInlineError = _ref5.showInlineError;
  return (
    /*#__PURE__*/
    (0, _core.css)("background-color:", theme.colors.destructive.foreground, ";color:", theme.colors.white, ";font-size:", theme.fontSizes[200], ";padding:", theme.spacing.xs, " ", theme.spacing.sm, ";position:relative;flex-grow:1;right:0;", showInlineError ? roundedBottom : detached, ";::before{content:'';display:block;width:0px;height:0px;border-left:5px solid transparent;border-right:5px solid transparent;border-bottom:5px solid ", theme.colors.destructive.foreground, ";position:absolute;top:-5px;right:10px;}", !showInlineError && {
      marginBottom: '0.5rem',
      transform: 'translate3d(0, 6px, 0)'
    }, " @media screen and (min-width:", theme.breakpoints.medium, "){position:absolute;", detached, ";user-select:none;z-index:1;", !active && (0, _styles.hideVisually)(), ";", showInlineError && inlineMessageOffset, ";}")
  );
};

var InlineError =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "e3hq9p23"
})("display:flex;justify-content:flex-end;position:relative;> *{", callout, ";}");
exports.InlineError = InlineError;
var CharacterLimit =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "e3hq9p24"
})("font-size:", function (_ref6) {
  var theme = _ref6.theme;
  return theme.fontSizes[200];
}, ";margin-top:", function (_ref7) {
  var theme = _ref7.theme;
  return theme.spacing.xs;
}, ";");
exports.CharacterLimit = CharacterLimit;
var ErrorTriangle =
/*#__PURE__*/
(0, _styled["default"])(_Icons.ErrorTriangle, {
  target: "e3hq9p25"
})("color:", function (_ref8) {
  var theme = _ref8.theme;
  return theme.colors.destructive.foreground;
}, ";");
exports.ErrorTriangle = ErrorTriangle;
var ScreenReaderOnly =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "e3hq9p26"
})((0, _styles.hideVisually)());
exports.ScreenReaderOnly = ScreenReaderOnly;
