// @ezcater/recipe/lib/components/EzField/EzSelect.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _react = _interopRequireWildcard(require("react"));

var _EzSelect = require("./EzSelect.styles");

var _hooks = require("../../utils/hooks");

var _EzCombobox = require("./EzCombobox");

var _EzTextInput = _interopRequireDefault(require("./EzTextInput"));

var _EzPopover = _interopRequireDefault(require("../EzPopover"));

var _Icons = require("../Icons");

var flatten = function flatten(options) {
  var grouped = new Map();
  options.forEach(function (item) {
    var group = item.group;
    var values = grouped.get(group) || [];
    values.push(item);
    grouped.set(group, values);
  });
  return (0, _toConsumableArray2["default"])(grouped);
};

var ListboxPopover = _react["default"].forwardRef(function (_ref, ref) {
  var targetRef = _ref.targetRef,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["targetRef"]);
  return _react["default"].createElement(_EzPopover["default"], {
    targetRef: targetRef,
    placement: "bottom-start",
    modifiers: [{
      name: 'matchWidth',
      enabled: true,
      fn: function fn(_ref2) {
        var state = _ref2.state;
        // eslint-disable-next-line no-param-reassign
        state.styles.popper.width = "".concat(state.rects.reference.width, "px");
      },
      phase: 'beforeWrite',
      requires: ['computeStyles']
    }, {
      name: 'offset',
      options: {
        offset: [0, 5]
      }
    }]
  }, _react["default"].createElement(_EzSelect.Listbox, (0, _extends2["default"])({
    role: "listbox",
    ref: ref
  }, props)));
});

var Option = function Option(_ref3) {
  var activeOption = _ref3.activeOption,
      activeOptionRef = _ref3.activeOptionRef,
      setActiveOption = _ref3.setActiveOption,
      option = _ref3.option,
      selected = _ref3.selected,
      onClick = _ref3.onClick;

  /* eslint-disable jsx-a11y/click-events-have-key-events */

  /* eslint-disable jsx-a11y/mouse-events-have-key-events */

  /* Note: lint doesn't detect the keyboard handler is on the input element, not the list */
  var id = (0, _hooks.useUniqueId)();
  var activeValue = activeOption && activeOption.value;
  return _react["default"].createElement("li", {
    role: "option",
    "aria-current": selected && option.value === selected.value || undefined,
    "aria-selected": activeValue === option.value,
    ref: activeValue === option.value ? activeOptionRef : undefined,
    onMouseOver: function onMouseOver() {
      return setActiveOption(option);
    },
    onClick: onClick,
    onMouseDown: function onMouseDown(e) {
      return e.preventDefault();
    } // used to prevent a focus event from bubbling up to the body in ie11
    ,
    id: id
  }, option.label);
};

var OptGroup = function OptGroup(props) {
  var id = (0, _hooks.useUniqueId)();

  var _props$group = (0, _slicedToArray2["default"])(props.group, 2),
      name = _props$group[0],
      options = _props$group[1];

  return _react["default"].createElement("li", null, _react["default"].createElement("ul", {
    role: "group",
    "aria-describedby": id
  }, _react["default"].createElement("li", {
    id: id,
    role: "presentation"
  }, name), options.map(function (o) {
    return _react["default"].createElement(Option, (0, _extends2["default"])({}, props, {
      option: o,
      key: o.label,
      onClick: function onClick() {
        return props.selectItem(o.value);
      }
    }));
  })));
};

var hasGroupedOptions = function hasGroupedOptions(options) {
  return options.some(function (o) {
    return o.group;
  });
};

var EzSelect = function EzSelect(props) {
  var options = props.options,
      value = props.value,
      onChange = props.onChange;
  var ariaLabelledBy = props['aria-labelledby'];
  var timeout = (0, _react.useRef)(null);
  var activeOptionRef = (0, _react.useRef)(null);
  var selected = options.find(function (o) {
    return o.value === value;
  });

  var _useState = (0, _react.useState)(selected),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      activeOption = _useState2[0],
      setActiveOption = _useState2[1];

  var setActiveIndex = function setActiveIndex(i) {
    return setActiveOption(i === -1 ? null : options[i]);
  };

  var listbox = {
    activeOption: activeOption,
    activeOptionRef: activeOptionRef,
    setActiveOption: setActiveOption,
    selected: selected
  };
  (0, _react.useEffect)(function () {
    return function () {
      return clearTimeout(timeout.current);
    };
  }, []);
  var comboboxState = (0, _EzCombobox.useComboboxState)();
  var hide = comboboxState.hide,
      visible = comboboxState.visible;

  var _useCombobox = (0, _EzCombobox.useCombobox)(comboboxState, {
    'aria-haspopup': 'listbox',
    className: props.className,
    disabled: props.disabled,
    error: props.error,
    touched: props.touched,
    opened: comboboxState.visible
  }),
      containerRef = _useCombobox.ref,
      optionsRef = _useCombobox.optionsRef,
      combobox = (0, _objectWithoutProperties2["default"])(_useCombobox, ["ref", "optionsRef"]);

  var handleKeyDown = function handleKeyDown(e) {
    var key = e.key;
    var activeIndex = options.indexOf(activeOption);

    var select = function select() {
      return selectItem(activeOption ? activeOption.value : null);
    };

    var prev = activeIndex <= 0 ? options.length - 1 : activeIndex - 1;
    var next = activeIndex === -1 || activeIndex >= options.length - 1 ? 0 : activeIndex + 1;
    var keyMap = {
      Escape: function Escape() {
        return setActiveIndex(-1);
      },
      ArrowUp: !e.defaultPrevented && function () {
        return setActiveIndex(prev);
      },
      ArrowDown: !e.defaultPrevented && function () {
        return setActiveIndex(next);
      },
      ' ': visible && select,
      Enter: visible && select
    };
    var action = keyMap[key];

    if (action) {
      e.preventDefault();
      action(e);
    }
  };

  (0, _hooks.useScrollIntoView)({
    containerRef: optionsRef,
    targetRef: activeOptionRef
  }, [activeOption, visible]);
  var comboboxInput = (0, _EzCombobox.useComboboxInput)(comboboxState, {
    'aria-autocomplete': 'list',
    'aria-labelledby': ariaLabelledBy,
    value: selected ? selected.label : '',
    'aria-activedescendant': !activeOptionRef.current ? '' : activeOptionRef.current.id,
    onKeyDown: handleKeyDown,
    onSelect: function onSelect(e) {
      return e.target.setSelectionRange(0, 0);
    },
    id: props.id,
    name: props.name,
    disabled: props.disabled,
    placeholder: props.placeholder,
    error: props.error,
    touched: props.touched,
    readOnly: true
  });
  var changeEvent = (0, _react.useCallback)(function (optionValue) {
    var event = new Event('change');
    containerRef.current.value = optionValue;
    containerRef.current.dispatchEvent(event);
    return event;
  }, [containerRef]);

  function selectItem(optionValue) {
    onChange(changeEvent(optionValue));
    timeout.current = setTimeout(hide, 100);
  }

  var comboboxFlyout = (0, _EzCombobox.useComboboxFlyout)(comboboxState);
  var move = (0, _react.useCallback)(function (option) {
    setActiveOption(option);
    if (visible) return;
    onChange(changeEvent(option.value));
  }, [visible, onChange, changeEvent]);
  (0, _hooks.useJumpToOption)(comboboxInput.ref, {
    options: options,
    move: move
  });
  return _react["default"].createElement(_EzSelect.Container, {
    ref: containerRef,
    hasError: props.touched && props.error,
    opened: visible
  }, _react["default"].createElement(_EzSelect.Combobox, combobox, _react["default"].createElement(_EzTextInput["default"], comboboxInput), _react["default"].createElement(_Icons.InsetIcon, {
    insetY0: true,
    right0: true,
    pr2: true
  }, _react["default"].createElement(_Icons.ChevronIcon, {
    flip: visible
  }))), visible && _react["default"].createElement(ListboxPopover, (0, _extends2["default"])({
    "aria-labelledby": ariaLabelledBy,
    targetRef: comboboxInput.ref
  }, comboboxFlyout, {
    ref: optionsRef,
    onClick: function onClick() {
      return comboboxInput.ref.current.focus();
    }
  }), hasGroupedOptions(options) ? _react["default"].createElement(_react["default"].Fragment, null, flatten(options).map(function (group) {
    return _react["default"].createElement(OptGroup, (0, _extends2["default"])({}, listbox, {
      group: group,
      key: group[0],
      selectItem: selectItem
    }));
  })) : options.map(function (o) {
    return _react["default"].createElement(Option, (0, _extends2["default"])({}, listbox, {
      option: o,
      key: o.label,
      onClick: function onClick() {
        return selectItem(o.value);
      }
    }));
  })));
};

var _default = EzSelect;
exports["default"] = _default;
