// @ezcater/recipe/lib/components/EzField/EzTimeInput.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _dayjs = _interopRequireDefault(require("dayjs"));

var _en = _interopRequireDefault(require("./en"));

var _hooks = require("../../utils/hooks");

var _EzTimeInput = require("./EzTimeInput.styles");

var _EzSelect = _interopRequireDefault(require("./EzSelect"));

var _Icons = require("../Icons");

var repeat = function repeat(n) {
  return new Array(n).fill(null);
};

var formatTime = function formatTime(time) {
  return time.toLocaleLowerCase().replace(/([0-9]{2})([a-z]{2})$/, '$1 $2');
};

var useTimeRangeOptions = function useTimeRangeOptions(_ref) {
  var start = _ref.start,
      end = _ref.end,
      step = _ref.step;

  var _useTranslation = (0, _hooks.useTranslation)(_en["default"]),
      t = _useTranslation.t;

  var date = (0, _dayjs["default"])().format(t('DATE_FORMAT'));
  var startTime = (0, _dayjs["default"])("".concat(date, " ").concat(formatTime(start)));
  var endTime = (0, _dayjs["default"])("".concat(date, " ").concat(formatTime(end)));
  var interval = Math.floor(endTime.diff(startTime, 'minute') / step) + 1;
  return repeat(interval).map(function (_, i) {
    return startTime.add(step * i, 'minute').format(t('TIME_FORMAT'));
  });
};

var _default = function _default(_ref2) {
  var start = _ref2.start,
      end = _ref2.end,
      _ref2$step = _ref2.step,
      step = _ref2$step === void 0 ? 60 : _ref2$step,
      value = _ref2.value,
      rest = (0, _objectWithoutProperties2["default"])(_ref2, ["start", "end", "step", "value"]);

  var _useTranslation2 = (0, _hooks.useTranslation)(_en["default"]),
      t = _useTranslation2.t;

  var date = (0, _dayjs["default"])().format(t('DATE_FORMAT'));
  var valueTime = (0, _dayjs["default"])("".concat(date, " ").concat(value));
  var valueTimeString = valueTime.format(t('TIME_FORMAT'));
  var error = rest.error,
      touched = rest.touched;
  var options = useTimeRangeOptions({
    start: start,
    end: end,
    step: step
  });
  return _react["default"].createElement(_EzTimeInput.TimeInput, null, _react["default"].createElement(_Icons.InsetIcon, {
    insetY0: true,
    left0: true,
    pl3: true
  }, _react["default"].createElement(_Icons.ClockIcon, null)), _react["default"].createElement(_EzSelect["default"], (0, _extends2["default"])({
    id: rest.id,
    label: rest.label
  }, {
    error: error,
    touched: touched
  }, {
    placeholder: rest.placeholder,
    options: options.map(function (option) {
      return {
        label: option,
        value: option
      };
    }),
    value: valueTimeString,
    onChange: rest.onChange,
    "aria-labelledby": rest['aria-labelledby']
  })));
};

exports["default"] = _default;
