// @ezcater/recipe/lib/components/EzField/Media.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _reactDom = require("react-dom");

var _core = require("@emotion/core");

var _cache = _interopRequireDefault(require("@emotion/cache"));

var _index = require("../index");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var sizes = {
  small: {
    width: 750,
    height: 1334
  },
  // phone e.g. iphone 6
  medium: {
    width: 1280,
    height: 800
  },
  // tablet e.g. amazon fire hd
  large: {
    width: 1920,
    height: 1080
  } // laptop e.g. dell xps 13

};

var Media = function Media(_ref) {
  var size = _ref.size,
      children = _ref.children;
  var iframeEl = (0, _react.useRef)(null);
  var linkRef = (0, _react.useRef)(null);

  var _useState = (0, _react.useState)(null),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      container = _useState2[0],
      setContainer = _useState2[1];

  var _useState3 = (0, _react.useState)(undefined),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      src = _useState4[0],
      setSrc = _useState4[1];

  var createStylesCache = (0, _react.useCallback)(function () {
    return (0, _cache["default"])({
      container: container.ownerDocument.head,
      key: 'frame'
    });
  }, [container]);
  (0, _react.useEffect)(function () {
    setContainer(iframeEl.current);
  }, [container]);
  return _react["default"].createElement("iframe", {
    title: size,
    frameBorder: "0",
    allowFullScreen: true,
    ref: iframeEl,
    style: _objectSpread({
      border: 'none',
      margin: 0
    }, sizes[size]),
    scrolling: "no",
    src: src
  }, iframeEl.current && !src && (0, _reactDom.createPortal)(_react["default"].createElement(_core.CacheProvider, {
    value: createStylesCache()
  }, _react["default"].createElement("link", {
    href: "https://fonts.googleapis.com/css?family=Lato:400,400i,700,700i&display=swap",
    rel: "stylesheet",
    ref: linkRef
  }), _react["default"].createElement(_index.EzGlobalStyles, null), children, linkRef.current && setSrc("data:text/html,".concat(encodeURIComponent("<html>".concat(linkRef.current.ownerDocument.head.outerHTML, "<body>").concat(iframeEl.current.innerHTML, "</body></html>"))))), iframeEl.current));
};

var _default = Media;
exports["default"] = _default;
