// reakit/lib/Disclosure/DisclosureContent.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var createComponent = require('reakit-system/createComponent');
var createHook = require('reakit-system/createHook');
var Box = require('../Box/Box.js');
var _rollupPluginBabelHelpers = require('../_rollupPluginBabelHelpers-770e06b1.js');
var React = require('react');
var useLiveRef = require('reakit-utils/useLiveRef');
require('reakit-utils/useSealedState');
var isSelfTarget = require('reakit-utils/isSelfTarget');
require('../Id/IdProvider.js');
require('../Id/IdState.js');
require('reakit-utils/useIsomorphicEffect');
var DisclosureState = require('./DisclosureState.js');

var useDisclosureContent = createHook.createHook({
  name: "DisclosureContent",
  compose: Box.useBox,
  useState: DisclosureState.useDisclosureState,
  useProps: function useProps(options, _ref) {
    var htmlOnTransitionEnd = _ref.onTransitionEnd,
        htmlOnAnimationEnd = _ref.onAnimationEnd,
        htmlStyle = _ref.style,
        htmlProps = _rollupPluginBabelHelpers._objectWithoutPropertiesLoose(_ref, ["onTransitionEnd", "onAnimationEnd", "style"]);

    var animating = options.animated && options.animating;

    var _React$useState = React.useState(null),
        transition = _React$useState[0],
        setTransition = _React$useState[1];

    var hidden = !options.visible && !animating;
    var style = hidden ? _rollupPluginBabelHelpers._objectSpread2({
      display: "none"
    }, htmlStyle) : htmlStyle;
    var onTransitionEndRef = useLiveRef.useLiveRef(htmlOnTransitionEnd);
    var onAnimationEndRef = useLiveRef.useLiveRef(htmlOnAnimationEnd);
    React.useEffect(function () {
      if (!options.animated) return undefined;
      var raf = window.requestAnimationFrame(function () {
        if (options.visible) {
          setTransition("enter");
        } else if (animating) {
          setTransition("leave");
        } else {
          setTransition(null);
        }
      });
      return function () {
        return window.cancelAnimationFrame(raf);
      };
    }, [options.animated, options.visible, animating]);
    var onEnd = React.useCallback(function (event) {
      if (!isSelfTarget.isSelfTarget(event)) return;
      if (!animating) return; // Ignores number animated

      if (options.animated === true) {
        var _options$stopAnimatio;

        (_options$stopAnimatio = options.stopAnimation) === null || _options$stopAnimatio === void 0 ? void 0 : _options$stopAnimatio.call(options);
      }
    }, [options.animated, animating, options.stopAnimation]);
    var onTransitionEnd = React.useCallback(function (event) {
      var _onTransitionEndRef$c;

      (_onTransitionEndRef$c = onTransitionEndRef.current) === null || _onTransitionEndRef$c === void 0 ? void 0 : _onTransitionEndRef$c.call(onTransitionEndRef, event);
      onEnd(event);
    }, [onEnd]);
    var onAnimationEnd = React.useCallback(function (event) {
      var _onAnimationEndRef$cu;

      (_onAnimationEndRef$cu = onAnimationEndRef.current) === null || _onAnimationEndRef$cu === void 0 ? void 0 : _onAnimationEndRef$cu.call(onAnimationEndRef, event);
      onEnd(event);
    }, [onEnd]);
    return _rollupPluginBabelHelpers._objectSpread2({
      id: options.baseId,
      "data-enter": transition === "enter" ? "" : undefined,
      "data-leave": transition === "leave" ? "" : undefined,
      onTransitionEnd: onTransitionEnd,
      onAnimationEnd: onAnimationEnd,
      hidden: hidden,
      style: style
    }, htmlProps);
  }
});
var DisclosureContent = createComponent.createComponent({
  as: "div",
  useHook: useDisclosureContent
});

exports.DisclosureContent = DisclosureContent;
exports.useDisclosureContent = useDisclosureContent;
