// @ezcater/recipe/lib/components/EzLink/EzLink.styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledAnchor = exports.StyledLink = void 0;

var _core = require("@emotion/core");

var _styled = _interopRequireDefault(require("../../themes/styled"));

var _styles = require("../../styles");

var base = function base(_ref) {
  var theme = _ref.theme;
  return (
    /*#__PURE__*/
    (0, _core.css)("font-weight:bold;text-decoration:none;", (0, _styles.pseudoClasses)('color', {
      color: theme.colors.interactive.base
    }), ";&:hover{text-decoration:underline;}")
  );
};

var link = function link(p) {
  return (
    /*#__PURE__*/
    (0, _core.css)("> *{", base(p), ";}")
  );
};

var StyledLink =
/*#__PURE__*/
(0, _styled["default"])("span", {
  target: "efnfn2t0"
})(link);
exports.StyledLink = StyledLink;
var StyledAnchor =
/*#__PURE__*/
(0, _styled["default"])("a", {
  target: "efnfn2t1"
})(base);
exports.StyledAnchor = StyledAnchor;
