// @ezcater/recipe/lib/components/EzButton/EzButton.styles.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.DisabledButtonWrapper = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _core = require("@emotion/core");

var _styledComponentVariant = _interopRequireDefault(require("styled-component-variant"));

var _styled = _interopRequireDefault(require("../../themes/styled"));

var _styles = require("../../styles");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var base = function base(_ref) {
  var theme = _ref.theme;
  return (
    /*#__PURE__*/
    (0, _core.css)("cursor:pointer;display:inline-block;outline:0;font-size:", theme.baseFontSize, ";font-family:", theme.baseFontFamily, ";font-weight:", theme.fontWeights.bold, ";line-height:1.25rem;min-height:1rem;text-decoration:none;user-select:none;white-space:nowrap;-webkit-tap-highlight-color:transparent;")
  );
};

var disabled = function disabled(_ref2) {
  var isDisabled = _ref2.disabled;
  return isDisabled &&
  /*#__PURE__*/
  (0, _core.css)("cursor:default;opacity:0.45;pointer-events:none;");
};

var outlineStyles = function outlineStyles(_ref3) {
  var theme = _ref3.theme;
  return (
    /*#__PURE__*/
    (0, _core.css)("border-radius:", theme.borderRadius[1], ";border-style:solid;border-width:", theme.borderWidth[0], ";box-shadow:0 1px 1px 0 rgba(0,0,0,0.12);padding:calc(", theme.spacing.xs, " - 0.125rem) ", theme.spacing.md, ";:focus{box-shadow:0px 0px 2px 2px ", theme.colors.interactive.focus.outline, ",0 1px 1px 0 rgba(0,0,0,0.12);}")
  );
};

var outline = (0, _styledComponentVariant["default"])('use', {
  primary: outlineStyles,
  secondary: outlineStyles
});

var primary = function primary(_ref4) {
  var colors = _ref4.theme.colors,
      destructive = _ref4.destructive;
  var color = destructive ? colors.destructive.foreground : colors.interactive.base;
  return (
    /*#__PURE__*/
    (0, _core.css)("color:", colors.white, ";border-color:", color, ";", (0, _styles.pseudoClasses)('border-color', {
      color: color,
      focusColor: (0, _styles.darken)(color, 0.25)
    }), ";", (0, _styles.pseudoClasses)('background-color', {
      color: color
    }), ";")
  );
};

var secondary = function secondary(_ref5) {
  var colors = _ref5.theme.colors,
      destructive = _ref5.destructive;
  var color = destructive ? colors.destructive.foreground : colors.interactive.base;
  var borderColor = colors.border.base;
  var focusColor = colors.interactive.focus.outline;
  return (
    /*#__PURE__*/
    (0, _core.css)("background-color:white;border:solid 1px;", (0, _styles.pseudoClasses)('border-color', {
      color: borderColor,
      focusColor: focusColor
    }), ";", (0, _styles.pseudoClasses)('color', {
      color: color
    }), ";&:hover:enabled{background-color:", colors.interactive.hover.background, ";}&:active:enabled{background-color:", colors.interactive.active.background, ";}")
  );
};

var tertiary = function tertiary(_ref6) {
  var colors = _ref6.theme.colors,
      destructive = _ref6.destructive;
  var color = destructive ? colors.destructive.foreground : colors.interactive.base;
  return (
    /*#__PURE__*/
    (0, _core.css)("background:none;border:none;line-height:1rem;padding:0;", (0, _styles.pseudoClasses)('color', {
      color: color
    }), ";&:focus{outline:auto;}&:hover{text-decoration:underline;}")
  );
};

var spinnerSize = 1.28571429;
var spinnerOptions = {
  size: "".concat(spinnerSize, "rem"),
  margin: "".concat(spinnerSize / 2, "rem"),
  thickness: '0.2rem',
  radius: '10rem'
};

var spinner = function spinner(_ref7) {
  var margin = _ref7.margin,
      size = _ref7.size,
      radius = _ref7.radius,
      thickness = _ref7.thickness,
      color = _ref7.color;
  return (
    /*#__PURE__*/
    (0, _core.css)("position:relative;color:transparent;&::before{position:absolute;content:'';top:50%;left:50%;margin:-", margin, " 0 0 -", margin, ";width:", size, ";height:", size, ";border-radius:", radius, ";border:", thickness, " solid rgba(0,0,0,0.15);}&::after{position:absolute;content:'';top:50%;left:50%;margin:-", margin, " 0 0 -", margin, ";width:", size, ";height:", size, ";animation:", _styles.keyframes.spin(), " 0.6s linear;animation-iteration-count:infinite;border-radius:", radius, ";border-color:", color, " transparent transparent;border-style:solid;border-width:", thickness, ";box-shadow:0 0 0 1px transparent;}")
  );
};

var loading = function loading(_ref8) {
  var isLoading = _ref8.isLoading,
      use = _ref8.use,
      colors = _ref8.theme.colors;
  var color = (0, _styledComponentVariant["default"])('use', {
    primary: colors.white,
    secondary: colors.grays[700]
  })({
    use: use
  });
  return isLoading && spinner(_objectSpread({}, spinnerOptions, {
    color: color
  }));
};

var use = (0, _styledComponentVariant["default"])('use', {
  primary: primary,
  secondary: secondary,
  tertiary: tertiary
});
var baseStyles = [base, use, disabled, loading, outline];
var InlineBlockWrapper =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "e1g91c7j0"
})("display:inline-block;");
var DisabledButtonWrapper = (0, _react.forwardRef)(function (props, ref) {
  return _react["default"].createElement(InlineBlockWrapper, (0, _extends2["default"])({
    ref: ref
  }, props), props.children);
});
exports.DisabledButtonWrapper = DisabledButtonWrapper;

var _default =
/*#__PURE__*/
(0, _styled["default"])('button', {
  target: "e1g91c7j1"
}).apply(void 0, baseStyles);

exports["default"] = _default;
