// reakit/lib/Popover/PopoverBackdrop.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var createComponent = require('reakit-system/createComponent');
var createHook = require('reakit-system/createHook');
require('../Box/Box.js');
var _rollupPluginBabelHelpers = require('../_rollupPluginBabelHelpers-770e06b1.js');
require('react');
require('reakit-utils/useLiveRef');
require('reakit-utils/useSealedState');
require('reakit-utils/isSelfTarget');
require('../Id/IdProvider.js');
require('../Id/IdState.js');
require('reakit-utils/useIsomorphicEffect');
require('../Disclosure/DisclosureState.js');
require('../Disclosure/DisclosureContent.js');
require('react-dom');
require('../Portal/Portal.js');
require('../Dialog/DialogState.js');
require('../DialogBackdropContext-9ef1477c.js');
var DialogBackdrop = require('../Dialog/DialogBackdrop.js');
require('@popperjs/core');
var PopoverState = require('./PopoverState.js');

var usePopoverBackdrop = createHook.createHook({
  name: "PopoverBackdrop",
  compose: DialogBackdrop.useDialogBackdrop,
  useState: PopoverState.usePopoverState,
  useOptions: function useOptions(_ref) {
    var _ref$modal = _ref.modal,
        modal = _ref$modal === void 0 ? false : _ref$modal,
        options = _rollupPluginBabelHelpers._objectWithoutPropertiesLoose(_ref, ["modal"]);

    return _rollupPluginBabelHelpers._objectSpread2({
      modal: modal
    }, options);
  }
});
var PopoverBackdrop = createComponent.createComponent({
  as: "div",
  memo: true,
  useHook: usePopoverBackdrop
});

exports.PopoverBackdrop = PopoverBackdrop;
exports.usePopoverBackdrop = usePopoverBackdrop;
