// @ezcater/recipe/lib/components/EzFlashMessage/EzFlashMessage.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _EzFlashMessage = require("./EzFlashMessage.styles");

var _CloseButton = _interopRequireDefault(require("../CloseButton"));

var _Icons = require("../Icons");

var icons = {
  error: _react["default"].createElement(_Icons.ErrorIcon, null),
  warning: _react["default"].createElement(_Icons.ErrorIcon, null),
  info: _react["default"].createElement(_Icons.InfoIcon, null),
  success: _react["default"].createElement(_Icons.SuccessIcon, null)
};
/**
 * Describe EzFlashMessage here.
 */

var EzFlashMessage = function EzFlashMessage(props) {
  return _react["default"].createElement(_EzFlashMessage.FlashMessageContainer, {
    use: props.use
  }, icons[props.use], _react["default"].createElement(_EzFlashMessage.FlashMessageContents, null, props.headline && _react["default"].createElement(_EzFlashMessage.FlashMessageHeadline, null, props.headline), props.children), props.onDismiss && _react["default"].createElement(_CloseButton["default"], {
    label: "Close",
    color: "red",
    onClick: props.onDismiss
  }));
};
/**
 * @component
 */


var _default = EzFlashMessage;
exports["default"] = _default;
