// @ezcater/recipe/lib/components/EzFlashMessage/EzFlashMessage.styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CloseButton = exports.FlashMessageHeadline = exports.FlashMessageContents = exports.FlashMessageContainer = void 0;

var _core = require("@emotion/core");

var _styledComponentVariant = _interopRequireDefault(require("styled-component-variant"));

var _styled = _interopRequireDefault(require("../../themes/styled"));

var coloredStyles = function coloredStyles(getColors) {
  return function (_ref) {
    var colors = _ref.theme.colors;

    var _getColors = getColors(colors),
        background = _getColors.background,
        foreground = _getColors.foreground,
        border = _getColors.border;

    return (
      /*#__PURE__*/
      (0, _core.css)("background-color:", background, ";border-left:5px solid ", foreground, ";box-shadow:inset -1px 0 0 ", border, ",inset 0 1px 0 ", border, ",inset 0 -1px 0 ", border, ";svg path{fill:", foreground, ";}")
    );
  };
};

var useStyles = (0, _styledComponentVariant["default"])('use', {
  error: coloredStyles(function (colors) {
    return colors.destructive;
  }),
  info: coloredStyles(function (colors) {
    return colors.info;
  }),
  success: coloredStyles(function (colors) {
    return colors.success;
  }),
  warning: coloredStyles(function (colors) {
    return colors.warning;
  })
});
var FlashMessageContainer =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "efgy8ud0"
})("border-radius:", function (props) {
  return props.theme.borderRadius[1];
}, ";border-width:1px;display:flex;padding:", function (props) {
  return props.theme.spacing.sm;
}, ";padding-right:", function (props) {
  return props.theme.spacing.lg;
}, ";", useStyles, ";p:last-of-type{margin-bottom:0;}");
exports.FlashMessageContainer = FlashMessageContainer;
var FlashMessageContents =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "efgy8ud1"
})("margin:0 ", function (props) {
  return props.theme.spacing.sm;
}, ";flex:1;");
exports.FlashMessageContents = FlashMessageContents;
var FlashMessageHeadline =
/*#__PURE__*/
(0, _styled["default"])("p", {
  target: "efgy8ud2"
})("margin-bottom:", function (props) {
  return props.theme.spacing.xs;
}, ";font-weight:bold;");
exports.FlashMessageHeadline = FlashMessageHeadline;
var CloseButton =
/*#__PURE__*/
(0, _styled["default"])("button", {
  target: "efgy8ud3"
})("border:0;background:none;outline:none;cursor:pointer;");
exports.CloseButton = CloseButton;
