// @ezcater/recipe/lib/components/EzLabel/EzLabel.styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _core = require("@emotion/core");

var _styledComponentVariant = _interopRequireDefault(require("styled-component-variant"));

var _styles = require("../../styles");

var _styled = _interopRequireDefault(require("../../themes/styled"));

var base = function base(_ref) {
  var theme = _ref.theme;
  return (
    /*#__PURE__*/
    (0, _core.css)("line-height:", theme.lineHeights.heading, ";font-weight:", theme.fontWeights.bold, ";padding:0;")
  );
};

var typography = function typography(_ref2) {
  var shade = _ref2.shade,
      size = _ref2.size;
  return function (_ref3) {
    var theme = _ref3.theme;
    return (
      /*#__PURE__*/
      (0, _core.css)("color:", theme.colors.text[shade], ";font-size:", theme.fontSizes[size], ";")
    );
  };
};

var size = (0, _styledComponentVariant["default"])('size', {
  small: typography({
    shade: 'deemphasis',
    size: 200
  }),
  normal: typography({
    shade: 'base',
    size: 300
  })
});

var top = function top(_ref4) {
  var theme = _ref4.theme;
  return {
    marginBottom: theme.spacing.xs2
  };
};

var bottom = function bottom(_ref5) {
  var theme = _ref5.theme;
  return {
    marginTop: theme.spacing.xs2
  };
};

var left = function left(_ref6) {
  var theme = _ref6.theme;
  return (
    /*#__PURE__*/
    (0, _core.css)("display:inline-block;margin-right:", theme.spacing.sm, ";")
  );
};

var right = function right(_ref7) {
  var theme = _ref7.theme;
  return (
    /*#__PURE__*/
    (0, _core.css)("display:inline-block;margin-left:", theme.spacing.sm, ";")
  );
};

var error = function error(_ref8) {
  var hasError = _ref8.error,
      theme = _ref8.theme;
  return hasError &&
  /*#__PURE__*/
  (0, _core.css)("color:", theme.colors.destructive.foreground, ";");
};

var position = (0, _styledComponentVariant["default"])('position', {
  top: top,
  bottom: bottom,
  left: left,
  right: right,
  hidden: _styles.hideVisually
});

var _default =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "eqn6k5a0"
})(base, position, size, error);

exports["default"] = _default;
