// @ezcater/recipe/lib/components/EzAlert/EzAlert.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _EzAlert = require("./EzAlert.styles");

var _EzTextStyle = _interopRequireDefault(require("../EzTextStyle"));

var _Icons = require("../Icons");

var icons = {
  error: _react["default"].createElement(_Icons.ErrorIcon, null),
  info: _react["default"].createElement(_Icons.InfoIcon, null),
  marketing: _react["default"].createElement(_Icons.MarketingIcon, null),
  success: _react["default"].createElement(_Icons.SuccessIcon, null),
  tip: _react["default"].createElement(_Icons.TipIcon, null),
  warning: _react["default"].createElement(_Icons.WarningIcon, null)
};

/**
 * Alerts represent highlighted messages on a page that call out important information.
 * Alerts can be used both with page content as well as a subheader as a status for the entire page.
 */
var EzAlert = function EzAlert(_ref) {
  var arrow = _ref.arrow,
      tagline = _ref.tagline,
      headline = _ref.headline,
      use = _ref.use;
  return _react["default"].createElement(_EzAlert.AlertContainer, {
    arrow: arrow,
    use: use
  }, icons[use], _react["default"].createElement(_EzAlert.AlertContent, null, _react["default"].createElement(_EzTextStyle["default"], {
    use: "strong"
  }, headline), _react["default"].createElement("div", null, tagline)));
};
/**
 * defaultProps
 * @property {string} use - Alerts are considered info by default.
 */


EzAlert.defaultProps = {
  use: 'info'
};
EzAlert.displayName = 'EzAlert';
/**
 * @component
 */

var _default = EzAlert;
exports["default"] = _default;
