// @ezcater/recipe/lib/components/EzAlert/EzAlert.styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AlertContent = exports.AlertContainer = void 0;

var _core = require("@emotion/core");

var _styledComponentVariant = _interopRequireDefault(require("styled-component-variant"));

var _styled = _interopRequireDefault(require("../../themes/styled"));

var sharedDirectionalStyles = function sharedDirectionalStyles() {
  return (
    /*#__PURE__*/
    (0, _core.css)("border-width:6px;border-style:solid;color:transparent;content:'';left:13px;position:absolute;")
  );
};

var bottomStyles = function bottomStyles() {
  return (
    /*#__PURE__*/
    (0, _core.css)("&::after{", sharedDirectionalStyles(), ";bottom:-12px;}")
  );
};

var topStyles = function topStyles() {
  return (
    /*#__PURE__*/
    (0, _core.css)("&::before{", sharedDirectionalStyles(), ";top:-12px;}")
  );
};

var coloredStyles = function coloredStyles(lightColor, darkColor) {
  return (
    /*#__PURE__*/
    (0, _core.css)("background-color:", lightColor, ";color:", darkColor, ";&::after{border-top-color:", lightColor, ";}&::before{border-bottom-color:", lightColor, ";}")
  );
};

var successStyles = function successStyles(_ref) {
  var colors = _ref.theme.colors;
  return (
    /*#__PURE__*/
    (0, _core.css)(coloredStyles(colors.success.background, colors.success.text), ";")
  );
};

var errorStyles = function errorStyles(_ref2) {
  var colors = _ref2.theme.colors;
  return (
    /*#__PURE__*/
    (0, _core.css)(coloredStyles(colors.destructive.background, colors.destructive.text), ";")
  );
};

var warningStyles = function warningStyles(_ref3) {
  var colors = _ref3.theme.colors;
  return (
    /*#__PURE__*/
    (0, _core.css)(coloredStyles(colors.warning.background, colors.warning.text), ";")
  );
};

var infoStyles = function infoStyles(_ref4) {
  var colors = _ref4.theme.colors;
  return (
    /*#__PURE__*/
    (0, _core.css)(coloredStyles(colors.info.background, colors.info.text), ";")
  );
};

var marketingStyles = function marketingStyles(_ref5) {
  var colors = _ref5.theme.colors;
  return (
    /*#__PURE__*/
    (0, _core.css)(coloredStyles(colors.teals.lighter, colors.teals.dark), ";")
  );
};

var tipStyles = function tipStyles(_ref6) {
  var colors = _ref6.theme.colors;
  return (
    /*#__PURE__*/
    (0, _core.css)(coloredStyles(colors.purples.lighter, colors.purples.dark), ";")
  );
};

var arrowStyles = (0, _styledComponentVariant["default"])('arrow', {
  bottom: bottomStyles,
  top: topStyles
});
var useStyles = (0, _styledComponentVariant["default"])('use', {
  error: errorStyles,
  tip: tipStyles,
  info: infoStyles,
  marketing: marketingStyles,
  success: successStyles,
  warning: warningStyles
});
var AlertContainer =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "etn6zcz0"
})("border-radius:", function (props) {
  return props.theme.borderRadius[1];
}, ";border-width:0;display:inline-flex;padding:", function (props) {
  return props.theme.spacing.sm;
}, ";padding-right:", function (props) {
  return props.theme.spacing.lg;
}, ";position:relative;", useStyles, ";", arrowStyles, ";");
exports.AlertContainer = AlertContainer;
var AlertContent =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "etn6zcz1"
})("margin-left:", function (props) {
  return props.theme.spacing.xs;
}, ";");
exports.AlertContent = AlertContent;
