// @ezcater/recipe/lib/components/EzOrderSummary/EzOrderSummary.styles.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Table", {
  enumerable: true,
  get: function get() {
    return _EzTable.Table;
  }
});
Object.defineProperty(exports, "Td", {
  enumerable: true,
  get: function get() {
    return _EzTable.Td;
  }
});
exports.SpecialInstructions = exports.SummarySection = exports.Total = exports.ItemsSection = void 0;

var _core = require("@emotion/core");

var _ = require("..");

var _EzTable = require("../EzTable/EzTable.styles");

var _styled = _interopRequireDefault(require("../../themes/styled"));

/*
  With table-layout: auto, widths of table cells only shrink to the size of their content,
  so setting the width to a size smaller than the content size causes the cell to always fit
  the content and not stretch to fill the available space.
*/
var cellFitContent = '1%';
var ItemsSection =
/*#__PURE__*/
(0, _styled["default"])(_EzTable.TableCardSection, {
  target: "ehxnl6n0"
})("&& table td{border:none;vertical-align:top;}&& table tr th,&& table tr td{padding:", function (_ref) {
  var theme = _ref.theme;
  return theme.spacing.sm;
}, " ", function (_ref2) {
  var theme = _ref2.theme;
  return theme.spacing.xs;
}, ";background-color:", function (_ref3) {
  var theme = _ref3.theme;
  return theme.colors.content.background;
}, ";}&& table tr th:first-of-type,&& table tr td:first-of-type{padding-left:", function (_ref4) {
  var theme = _ref4.theme;
  return theme.spacing.xs;
}, ";}&& table tr th:last-of-type,&& table tr td:last-of-type{padding-right:", function (_ref5) {
  var theme = _ref5.theme;
  return theme.spacing.lg;
}, ";}th:first-of-type,td:first-of-type{width:", function (_ref6) {
  var theme = _ref6.theme;
  return theme.spacing.xl4;
}, ";}td:last-of-type{width:", cellFitContent, ";}");
exports.ItemsSection = ItemsSection;
var Total =
/*#__PURE__*/
(0, _styled["default"])("span", {
  target: "ehxnl6n1"
})("font-size:", function (_ref7) {
  var theme = _ref7.theme;
  return theme.fontSizes[500];
}, ";font-weight:", function (_ref8) {
  var theme = _ref8.theme;
  return theme.fontWeights.bold;
}, ";"); // left padding here is the fixed width of the first column of the previous table
// plus the padding of the second column

exports.Total = Total;

var totals = function totals(_ref9) {
  var theme = _ref9.theme;
  return (
    /*#__PURE__*/
    (0, _core.css)("padding-left:calc(", theme.spacing.xl4, " + ", theme.spacing.xs, ");table{width:auto;margin-left:calc(", theme.spacing.xs, " * -1);margin-right:calc(", theme.spacing.xs, " * -1);}th,td{text-align:left;font-weight:normal;width:100%;border:none;padding:", theme.spacing.xs2, " ", theme.spacing.xs, ";}td:last-of-type{text-align:right;}tr:nth-of-type(even){background-color:", theme.colors.grays[100], ";}")
  );
};

var SummarySection =
/*#__PURE__*/
(0, _styled["default"])(_.EzCardSection, {
  target: "ehxnl6n2"
})(totals);
exports.SummarySection = SummarySection;

var tableware = function tableware(_ref10) {
  var theme = _ref10.theme;
  return (
    /*#__PURE__*/
    (0, _core.css)("background-color:", theme.colors.grays[100], ";border-radius:", theme.borderRadius[1], ";display:inline-block;font-size:", theme.fontSizes[200], ";margin-top:", theme.spacing.xs, ";padding:", theme.spacing.sm, " ", theme.spacing.md, ";")
  );
};

var SpecialInstructions =
/*#__PURE__*/
(0, _styled["default"])("div", {
  target: "ehxnl6n3"
})(tableware); // EzCard checks for a card section (by displayName), and will wrap if it can't find one

exports.SpecialInstructions = SpecialInstructions;
ItemsSection.displayName = _.EzCardSection.displayName;
SummarySection.displayName = _.EzCardSection.displayName;
