// @ezcater/recipe/lib/components/EzOrderSummary/EzOrderSummary.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _react = _interopRequireDefault(require("react"));

var _EzOrderSummary = require("./EzOrderSummary.styles");

var _ = require("..");

var _en = _interopRequireDefault(require("./en"));

var _hooks = require("../../utils/hooks");

var Item = function Item(_ref) {
  var item = _ref.item;

  var _useTranslation = (0, _hooks.useTranslation)(_en["default"]),
      t = _useTranslation.t;

  return _react["default"].createElement("div", null, _react["default"].createElement("div", null, item.name ? _react["default"].createElement("span", null, item.name, " @ ", item.price) : _react["default"].createElement("span", null, t('Tableware'))), item.options && item.options.map(function (option, index) {
    return _react["default"].createElement("div", {
      key: index
    }, _react["default"].createElement(_.EzTextStyle, {
      use: "subdued"
    }, option.label, ": ", option.value));
  }), item.specialInstructions && _react["default"].createElement(_EzOrderSummary.SpecialInstructions, null, _react["default"].createElement("div", null, _react["default"].createElement(_.EzTextStyle, {
    use: "strong"
  }, t('Special Instructions:'))), item.specialInstructions));
};
/**
 * An order summary is an at-a-glance breakdown of billable items that make up an order.
 */


var EzOrderSummary = function EzOrderSummary(_ref2) {
  var actions = _ref2.actions,
      items = _ref2.items,
      subtitle = _ref2.subtitle,
      tableware = _ref2.tableware,
      title = _ref2.title,
      summary = _ref2.summary;

  var _useTranslation2 = (0, _hooks.useTranslation)(_en["default"]),
      t = _useTranslation2.t;

  return _react["default"].createElement(_.EzCard, {
    actions: actions,
    title: title,
    subtitle: subtitle
  }, _react["default"].createElement(_EzOrderSummary.ItemsSection, null, _react["default"].createElement(_.EzTable, {
    columns: [{
      heading: t('Qty'),
      accessor: 'quantity',
      numeric: true
    }, {
      heading: t('Item'),
      accessor: Item
    }, {
      heading: t('Price'),
      accessor: 'total',
      numeric: true
    }],
    items: [].concat((0, _toConsumableArray2["default"])(items), [tableware])
  })), _react["default"].createElement(_EzOrderSummary.SummarySection, null, _react["default"].createElement("table", null, _react["default"].createElement("tbody", null, summary.lineItems.map(function (subtotal) {
    return _react["default"].createElement("tr", {
      key: subtotal.label
    }, _react["default"].createElement("th", null, subtotal.label), _react["default"].createElement("td", null, subtotal.value));
  }))), _react["default"].createElement("div", null, _react["default"].createElement(_.EzLayout, {
    layout: "split"
  }, _react["default"].createElement(_EzOrderSummary.Total, null, t('Total')), _react["default"].createElement(_EzOrderSummary.Total, null, summary.total)), summary.perHead && _react["default"].createElement(_.EzLayout, {
    layout: "split"
  }, _react["default"].createElement(_.EzTextStyle, {
    use: "subdued"
  }, t('Price per head')), _react["default"].createElement(_.EzTextStyle, {
    use: "subdued"
  }, t('{{perHead}}/person', {
    perHead: summary.perHead
  }))))));
};

var _default = EzOrderSummary;
exports["default"] = _default;
